src/interface.jl

```diff
@@ -70,6 +70,8 @@ function _laplacetype end
 _valtype(basis::AbstractPoly4MLBasis, x::SINGLE) = 
       _valtype(basis, typeof(x)) 
 
+_valtype(basis::ScalarPoly4MLBasis, ::Type{<: StaticVector{3, S}}) where {S <: Real} = _valtype(basis, S)
```

Comment:
I don't understand this. `ScalarP4MLBasis` should only accept scalars as input, not vectors. Am I misunderstanding something?

Reply:
what is worse is that the type restriction to be a 3d vector is increadibly strong. This should not be needed.

Reply:
I did this because we need the following general `rrule` for each embedding:

```
ChainRulesCore.rrule(evaluate, embedding, X)
```

But I think you are right - on the second thought maybe we can do something like 

```
struct SVecScalarEmbedding
   TB ::ScalarP4MLBasis
end
```

and then implement:

```
ChainRulesCore.rrule(evaluate, embedding::SVecScalarEmebdding, X)
```

?

Reply:
No I still don't understand. There should simply be a layer in front of the radial  basis that takes a particle and returns the norm.

Reply:
hmm I think if we have `Zygote` as dependency then it is clear to me on how to do that. But with only `ChainRulesCore` maybe I have to give it thought how to do this so that `rrule` can be called correctly...

Reply:
The radial embedding must be allowed to an arbitrary chain.

Reply:
Can you point out where in the code this is a problem?